### PR TITLE
add docstring to ax.quiver

### PR DESCRIPTION
**Pull request by K-Monty**
## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by tacaswell** [comment]
Ah, this PR replaces #19562.  Copying my comment from that PR https://github.com/matplotlib/matplotlib/pull/19562#issuecomment-786112939:

----

@K-Monty Sorry for the slow response.

This did not work because in https://github.com/matplotlib/matplotlib/blob/a510e32a37a8b4e0bc4e8384df20c7b45abd9913/lib/matplotlib/axes/_axes.py#L5100 we set the `fun.__doc__` attribute directly which overrides the one you added in this PR!  If you are still interested in working on this you need to edit the string in `mquiver.Quiver.quiver_doc`

**Comment by K-Monty** [comment]
@tacaswell Thanks for your feedback! I forgot to remove that line (quiver.__doc__ = mquiver.Quiver.quiver_doc ), but my original intention was to separate the documentation in mquiver.Quiver.quiver_doc and the one I edited, so that they are independent of each other (mquiver.Quiver.quiver_doc will still be without 'return'). 
But I am not sure whether this is redundant, and I should just edit the documentation in mquiver.Quiver.quiver_doc, as you suggested?

**Comment by tacaswell** [comment]
Please edit the existing string. I think we use it in multiple places and want to keep them in sync with each other.

File: lib/matplotlib/quiver.py [hunk 1]
```diff
@@ -186,6 +186,10 @@
 
     %(PolyCollection_kwdoc)s
 
+Returns
+----------------
+The new `matplotlib.quiver.Quiver` object.
```
Comment (timhoffm) [inline lib/matplotlib/quiver.py:191]:
```suggestion
-------
matplotlib.quiver.Quiver
```
See https://numpydoc.readthedocs.io/en/latest/format.html#sections on the `Returns` section.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -7,12 +7,12 @@
 import numpy as np
 from numpy import ma
 
-import matplotlib.category  # Register category unit converter as side-effect.
+#import matplotlib.category  # Register category unit converter as side-effect.
```
Comment (timhoffm) [inline lib/matplotlib/axes/_axes.py:10]: Please revert commenting out these imports. As the comment states, we need them for their side-effects.

Comment (K-Monty) [reply lib/matplotlib/axes/_axes.py:10]: Thanks for your feedback! I will make the changes as requested.

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @K-Monty! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Comment by K-Monty** [comment]
Thanks a lot, guys! 😊

**Pull request merged**